src/mono/wasm/debugger/BrowserDebugProxy/Firefox/FirefoxDebuggerProxy.cs

```diff
@@ -26,10 +28,13 @@ public static void StartListener(int proxyPort, ILogger logger, int browserPort
     {
         if (s_tcpListener is null)
         {
+            //when running from blazor always try to open the same port(which is sent by parameter) to avoid creating a lot of remote debugging connections on firefox
+            if (IPGlobalProperties.GetIPGlobalProperties().GetActiveTcpListeners().Any(x => x.Port == proxyPort))
+                proxyPort = 0;
```

Comment:
Maybe:

```suggestion
            // If there is an existing listener on @proxyPort, then use a new dynamic port.
            // Blazor always tries to open the same port (specified in @proxyPort) to avoid
            // creating a lot of remote debugging connections on firefox
            if (IPGlobalProperties.GetIPGlobalProperties().GetActiveTcpListeners().Any(x => x.Port == proxyPort))
            {
                proxyPort = 0;
            }
```

Reply:
What's the use case where we have valid `proxyPort` argument, and there is somebody listening on the port, *and* the user wants to automatically switch to a different port?

Reply:
When we press alt-shift-d on firefox, I prefer to use the port that I'm passing as parameter which is 6001, but if it's not available it can use any other, on Blazor side we will get the listening port printed on console and the debugging will work.

Reply:
Is that a condition that would never happen with blazor? Or if it does, then does that indicate that a proxy might be stuck? Would blazor prefer for us to fail in that case, or fallback to using a new port?

And the same thing for non-blazor wasm projects - should we fail if get an explicit port, and using a dynamic port should be an explicit action, say, with a command line argument?

Reply:
I think that this may happen, blazor is passing port 6001 as parameter, and the port is in use for other app, not related with blazor. Then it will listen on another port and print the dynamic port. 
This doesn't indicate that the proxy might be stuck, I think browserdebugproxy never gets stuck as it's a process linked to the aspnetcore app, and then the aspnetcore app is finalized browserdebugproxy it's automatically killed.
Do you want a parameter like --autoTryAnotherPort and only try another port if this parameter is passed?

Reply:
@javiercn what do you think about this? If the requested proxy port is in use, then should we always automatically fallback to a dynamic port?